### PR TITLE
Inline Class Documentation

### PR DESCRIPTION
**Pull request by SIsilicon**
This pull requests aims to document the classes added by this project. That way, users can find references to the API from within the editor, just as how they would with builtin classes.
image

Here's an example of how the documentation works.
```gdscript
## The first line is a brief description.
## Subsequent lines are part of the full description. All comments that start with "##" are considered document strings.
## Each class must be put before, or inline with certain keywords.
## In this case, this document block should be placed immediately above tool, extends, or class_name to be considered general class documentation.
## A document block may have annotations in each line. Each annotation has a particular meaning.
## 
## @tutorial https://link.to.tutorial
## @tutorial http://you.can.have/multiple
## @contribute https://link.to.where.you.contribute.documentation
## @meta-key value that metadata "key" will store (author, revision, etc...)
## 
## Use "@doc-ignore" if you don't want the script showing up in documentation.
## Also, the format used here is [i]special bbcode[/i]. That was just italized. [code]This is code.[/code]
extends Reference
class_name Example

## Nothing special
const CONSTANT = "r"

# If enumerator isn't named, its contents will be treated like individual constants.
# No, this isn't a doc string.
enum Enum{
    ENUM_A ## Description of ENUM_A
}

## @type var_type
## This is optional
##
## @default 0.0
## This needs to be defines if your variable has an initial value.
## 
## @enum SomeOtherClass.Enum
## If your variable holds an enum, define it here. If the enum were defines here, it could just be named directly without the dot.
##
## @setter set_var
## @getter get_var
## 
## This documentation will be for the variable defined below.
var foo := 0.0

var bar := 0 ## Documentation can be on the same line.

# variables and methods with "_" prepended are assumed to be private and ignored unless a doc string is defined before or alongside it.
var _private = null


## @args a, b, c, d
## Naming them is mandatory.
##
## @arg-types int, int, Spatial, Variant
## @arg-enums Enum
## @return int
## @return-enum Enum
## Use "@virtual" if the function is to be overwritten.
func something(a: Enum.ENUM_A, b: int, c: Spatial, d) -> int:
    return Enum.ENUM_A

## @arg-types float, int
## @arg-enums , Enum
signal some_signal(a, b)

```

Closes https://github.com/GodotNuts/GodotFirebase/issues/116.

### CODE REVIEW

File: addons/godot-firebase/firebase/firebase.gd [hunk 1]
```diff
@@ -27,23 +46,23 @@ var config : Dictionary = {
...
-    Auth.set_config(config)
-    Firestore.set_config(config)
-    Database.set_config(config)
-    Storage.set_config(config)
-    DynamicLinks.set_config(config)
+    _load_config()
+    Auth._set_config(_config)
+    Firestore._set_config(_config)
+    Database._set_config(_config)
+    Storage._set_config(_config)
```
Comment (WolfgangSenff) [inline addons/godot-firebase/firebase/firebase.gd:65]: Apparently this is missing the DynamicLinks._set_config(_config).

File: addons/silicon.util.custom_docs/doc_exporter/editor_help_doc_exporter.gd [hunk 2]
```diff
@@ -0,0 +1,1046 @@
...
+			any_previous = !m.empty()
+		
+		label.pop() #table
+		label.pop()
+		label.pop() # font
+		label.add_text("\n")
+		label.add_text("\n")
+
+	# Theme properties
+#	if doc.theme_properties.size():
```
Comment (WolfgangSenff) [inline addons/silicon.util.custom_docs/doc_exporter/editor_help_doc_exporter.gd:330]: We should get rid of all the commented out code, if it's not going to be used.

**Review by WolfgangSenff — CHANGES_REQUESTED** [review]

File: addons/silicon.util.custom_docs/doc_exporter/editor_help_doc_exporter.gd (hunk 2 above)
Comment (SIsilicon) [reply addons/silicon.util.custom_docs/doc_exporter/editor_help_doc_exporter.gd:330]: In terms of that plugin, I might expose theme properties in the future, so I'm just leaving the commented code there until I decide to implement it.

File: addons/godot-firebase/firebase/firebase.gd (hunk 1 above)
Comment (SIsilicon) [reply addons/godot-firebase/firebase/firebase.gd:65]: Fixed!

**Pull request merged**